cmd/kubelet/app/server.go

```diff
@@ -205,6 +209,12 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 				glog.Fatal(err)
 			}
 
+			// We have validated everything that is side-effect free, exit now
```

Comment:
Why not also include the currently-referenced dynamic config file if it exists?
If admins are trying to use this in live environments, I'd worry that it could cause confusion not to check the full stack of config sources.
As you said, bootstrapping the controller is not pure, so maybe that's a valid justification.
OTOH, Bootstrap() doesn't do *that* much beyond setting up the dynamic-config-dir (when it does not exist) and trying to read/validate the config in it. There is the goroutine it launches to check the trial duration, but the other Kubelet process is potentially doing this anyway, and I wouldn't expect this Kubelet to behave differently given the same config/flags. Kubelet should be writing things atomically...
But, eh, this is speculation and side-effects are scary...